src/testWithSpringBoot_2_4/java/org/openrewrite/java/spring/MigrateHandlerInterceptorTest.java

```diff
@@ -66,4 +67,54 @@ public boolean preHandle(HttpServletRequest request, HttpServletResponse respons
           )
         );
     }
+
+    @Issue("https://github.com/openrewrite/rewrite-spring/issues/620")
+    @Test
+    void doesNotReplaceInterceptorsExtendingOwnInterceptors() {
+        //language=java
+        rewriteRun(
+          // Do change classes that directly extend HandlerInterceptorAdapter
+          java(
+            """
+              import javax.servlet.http.*;
+
+              import org.springframework.web.servlet.handler.HandlerInterceptorAdapter;
+
+              class MySuperInterceptor extends HandlerInterceptorAdapter {
+                  @Override
+                  public boolean preHandle(HttpServletRequest request, HttpServletResponse response, Object handler) {
+                      return super.preHandle(request, response, handler);
+                  }
+              }
+              """,
+            """
+              import javax.servlet.http.*;
+
+              import org.springframework.web.servlet.HandlerInterceptor;
+
+              class MySuperInterceptor implements HandlerInterceptor {
+                  @Override
+                  public boolean preHandle(HttpServletRequest request, HttpServletResponse response, Object handler) {
+                      return HandlerInterceptor.super.preHandle(request, response, handler);
```

Comment:
In your comments you pointed out that this should be 
```suggestion
                      return true;
```
And looking at [HandlerInterceptor.java#L103-L107](https://github.com/spring-projects/spring-framework/blob/80b72151a7984b5151311d93c59e0f46b61c2c50/spring-webmvc/src/main/java/org/springframework/web/servlet/HandlerInterceptor.java#L103-L107) I tend to agree; what made you not make that change just yet?

Reply:
There is a difference between bugfixing and changing the intention of things. I tend to separate the two, unless discussed. As you agree that the intention of this recipe should be changed, I will make a change accordingly.